bundles/framework/layerlist/view/LayerViewTabs/LayerList/LayerCollapse/LayerCollapseHandler.js

```diff
@@ -82,16 +85,82 @@ class ViewHandler extends StateHandler {
         setTimeout(() => this.sandbox.postRequestByName('RemoveMapLayerRequest', [id]), ANIMATION_TIMEOUT);
     }
 
+    // Check if we need to show confirm dialog when activating a whole group
+    showWarn (group) {
+        if (group.layers.length > 9) {
+            if (!group.id || this.state.showWarn.includes(group.id)) {
+                return;
+            }
+            const showWarn = [...this.state.showWarn, group.id];
+            // set groups id to showWarn array so we show the confirm dialog
+            this.updateState({ showWarn });
+        } else {
+            return false;
+        }
+    }
+
+    deactivateGroup (group) {
+        if (this.state.showWarn.includes(group.id)) {
+            const index = this.state.showWarn.indexOf(group.id);
+            if (index === -1) {
+                return;
+            }
+            const showWarn = [...this.state.showWarn];
+            showWarn.splice(index, 1);
+            this.updateState({ showWarn });
+        }
+    }
```

Comment:
I don't see a reason why you should need to track the warning like this instead of just wrapping the checkbox with a confirm tag if there are too many layers that will be added to the map. This makes the API more difficult to understand than it needs to be and you need to do bookkeeping that you don't need to do.

Reply:
Could the logic be changed so that the checkbox is _not_ wrapped inside the confirm tag if there's less than x layers that would be added?